main.go

```diff
@@ -0,0 +1,54 @@
+package main
+
+import (
+	"encoding/csv"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+
+	"github.com/gorilla/mux"
+)
+
+type Pokemon struct {
+	Id      int    `json:"id"`
+	Name    string `json:"name"`
+	TypeOne string `json:"typeOne"`
+}
+
+func main() {
+
+	pokemons := initializePokemonData()
+
+	r := mux.NewRouter()
+	r.HandleFunc("/pokemon/{id}", func(rw http.ResponseWriter, r *http.Request) {
+		HandlePokemonById(rw, r, pokemons)
+	})
+	http.ListenAndServe(":8080", r)
+}
+
+func initializePokemonData() []Pokemon {
+	csvfile, err := os.Open("pokemons.csv")
```

Comment:
You would probably like to avoid hardcoding this path as well. It wouldn't be a bad idea to get the path as a function param, or from a config file